### PR TITLE
Add math.tan expansion to polynomial approximation

### PR DESCRIPTION
**Pull request by rsuderman**
We can use the math.tan expansion with the polynomial approximations of math.sin and math.cos to estimate the value of math.tan.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/test/BUILD [hunk 1]
```diff
@@ -38,6 +38,7 @@ iree_lit_test_suite(
             "iree_comprehensive_bufferize.mlir",
             "lower_ukernel_to_calls.mlir",
             "pad_dynamic_alloc.mlir",
+            "polynomial-approximation.mlir",
```
Comment (benvanik) [inline compiler/src/iree/compiler/Codegen/Common/test/BUILD]: style: use _ like the rest of the tests

File: compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// RUN: iree-opt --iree-codegen-polynomial-approximation %s | FileCheck %s
+
+// CHECK-LABEL: @polynomial_tan
+func.func @polynomial_tan(%arg0: f32) -> f32 {
+    // CHECK-NOT: math.tan
+    %0 = math.tan %arg0 : f32 
+    return %0 : f32
+}
```
Comment (benvanik) [inline compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir]: style: newlines at ends of files (usually an editor setting)

File: compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// RUN: iree-opt --iree-codegen-polynomial-approximation %s | FileCheck %s
+
+// CHECK-LABEL: @polynomial_tan
+func.func @polynomial_tan(%arg0: f32) -> f32 {
+    // CHECK-NOT: math.tan
```
Comment (benvanik) [inline compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir]: style: 2 space indent

File: compiler/src/iree/compiler/Codegen/Common/test/BUILD (hunk 1 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Codegen/Common/test/BUILD]: Done.

File: compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir (hunk 2 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir]: Done.

File: compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir (hunk 3 above)
Comment (rsuderman) [reply compiler/src/iree/compiler/Codegen/Common/test/polynomial-approximation.mlir]: Done.

**Review by benvanik — APPROVED** [review]
thanks!
(may be worth a benchmark run? I'd expect this to be faster if we have any models in the corpus using tan, but I'm not sure we do)

**Comment by rsuderman** [comment]
> thanks! (may be worth a benchmark run? I'd expect this to be faster if we have any models in the corpus using tan, but I'm not sure we do)

In this case it let them run at all - the current state just failed being unable to link to libm.

**Comment by benvanik** [comment]
hah! well, that libm route would have been stupid slow scalarized nastiness, so let's just call this 1000x faster :P

**Pull request merged**